pymc3/distributions/posterior_predictive.py

```diff
@@ -302,12 +297,10 @@ def extend_trace(self, trace: Dict[str, np.ndarray]) -> None:
             except KeyboardInterrupt:
                 pass
 
-        if keep_size:
-            return {
-                k: ary.reshape((nchains, ndraws, *ary.shape[1:])) for k, ary in ppc_trace.items()
-            }
-        # this gets us a Dict[str, np.ndarray] instead of my wrapped equiv.
-        return ppc_trace.data
+    if keep_size:
+        return {k: ary.reshape((nchains, ndraws, *ary.shape[1:])) for k, ary in ppc_trace.items()}
+    # this gets us a Dict[str, np.ndarray] instead of my wrapped equiv.
+    return ppc_trace.data
```

Comment:
moving this outside from the context manager so mypy recognises that the return statement is always reached